crates/router/src/connector/bankofamerica/transformers.rs

```diff
@@ -2116,3 +2655,28 @@ impl
         }
     }
 }
+
+fn get_commerce_indicator(network: Option<String>) -> String {
+    match network {
+        Some(card_network) => match card_network.to_lowercase().as_str() {
+            "amex" => "aesk",
+            "discover" => "dipb",
+            "mastercard" => "spa",
+            "visa" => "internet",
+            _ => "internet",
+        },
+        None => "internet",
+    }
+    .to_string()
+}
+
+fn is_setup_mandate_payment(item: &types::CompleteAuthorizeData) -> bool {
+    matches!(item.amount, 0) && is_merchant_initiated_mandate_payment(item)
+}
+
+fn is_merchant_initiated_mandate_payment(item: &types::CompleteAuthorizeData) -> bool {
+    item.setup_future_usage.map_or(false, |future_usage| {
+        matches!(future_usage, common_enums::FutureUsage::OffSession)
+    })
```

Comment:
These conditions are not for merchant initiated mandate payment


```suggestion
fn is_customer_initiated_mandate_payment(item: &types::CompleteAuthorizeData) -> bool {
    item.setup_future_usage.map_or(false, |future_usage| {
        matches!(future_usage, common_enums::FutureUsage::OffSession)
    }) && (item.request.customer_acceptance.is_some()
                || item
                    .request
                    .setup_mandate_details
                    .clone()
                    .map_or(false, |mandate_details| {
                        mandate_details.customer_acceptance.is_some()
                    }))
```

Reply:
Also if possible use this wherever same parameters are being checked